sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageExtensions.cs

```diff
@@ -26,9 +27,21 @@ private static IEnumerable<Data> AsAmqpData(this IEnumerable<ReadOnlyMemory<byte
         {
             foreach (ReadOnlyMemory<byte> data in binaryData)
             {
+                ArraySegment<byte> segment;
+                if (!data.IsEmpty)
+                {
+                    if (!MemoryMarshal.TryGetArray(data, out segment))
+                    {
+                        segment = new ArraySegment<byte>(data.ToArray());
+                    }
+                }
+                else
```

Comment:
I wonder if we could streamline into something like:

```csharp
private static IEnumerable<Data> AsAmqpData(this IEnumerable<ReadOnlyMemory<byte>> binaryData)
{
    foreach (ReadOnlyMemory<byte> data in binaryData)
    {
        ArraySegment<byte> segment;
        
        if (!MemoryMarshal.TryGetArray(data, out segment))
        {
            // Not the actual exception, of course.
            throw new Exception("OMG WTF?");  
        }
        
        yield return new Data
        {
            Value = segment
        };
    }
}
```

Reply:
- `null` would return an empy array segment
- empty ROM would return an empty array segment
- populated ROM would return the populated array segment
- populated ROM that could not be converted to `byte` would throw

Reply:
I don't think we need to throw when MemoryMarshal fails - we use this same strategy in [BinaryData](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/System.Memory.Data/src/BinaryData.cs#L205)